### PR TITLE
Issue #2710 - update to use FATRunner and the standard FATServlet

### LINKED ISSUE #2710: MP Config: Use FATRunner in the Config FAT Suite
Tom found a `FULL` test running in the continuous build and we worked out this was caused by not using `@RunWith(FATRunner.class)`.

We should probably use it on most of our tests because it gives us nice things (like names when using `RepeatTests`)

### PR DESCRIPTION
**Pull request by tevans78**
fixes #2710 
#build

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/CDIBrokenInjectionTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addPackages(true, "com.ibm.ws.microprofile.appConfig.cdi.broken")
+                        .addAsManifestResource(new File("test-applications/" + APP_NAME + ".war/resources/META-INF/permissions.xml"), "permissions.xml")
+                        .addAsLibrary(SharedShrinkWrapApps.cdiConfigJar());
+
+        ShrinkHelper.exportDropinAppToServer(server, war);
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/CDIBrokenInjectionTest.java]: May need to be aware that this exports to the `publish` directory as well as to the server install. If the server is reused, the app will still be deployed.

File: dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SysPropServlet.java [hunk 2]
```diff
@@ -12,26 +12,52 @@
...
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.ibm.ws.microprofile.appConfig.cdi.beans.SessionScopedConfigFieldInjectionBean;
+
 @SuppressWarnings("serial")
 @WebServlet("/system")
 public class SysPropServlet extends HttpServlet {
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SysPropServlet.java]: This seems to be doing a whole bunch more than it used to...

File: dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SessionScopedFieldTestServlet.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 @SuppressWarnings("serial")
 @WebServlet("/fieldSession")
-public class SessionScopedFieldTestServlet extends AbstractBeanServlet {
+public class SessionScopedFieldTestServlet extends FieldTestServlet {
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SessionScopedFieldTestServlet.java]: This doesn't seem to be used any more. Can it be removed?

File: dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SysPropServlet.java (hunk 2 above)
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SysPropServlet.java]: Did you collapse `SessionScopedFieldTestServlet` into this class? Can it be removed?

File: dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/CDIBrokenInjectionTest.java (hunk 1 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/CDIBrokenInjectionTest.java]: Server is only used once

File: dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/CDIBuiltInConverterTest.java [hunk 4]
```diff
@@ -10,173 +10,54 @@
...
  */
-public class CDIBuiltInConverterTest extends LoggingTest {
-
-    @ClassRule
-    public static SharedServer SHARED_SERVER = new ShrinkWrapSharedServer("CDIConfigServer");
-
-    @ClassRule
-    public static RepeatTests r = RepeatTests
-                    .with(RepeatConfig11EE7.INSTANCE.forServers(SHARED_SERVER.getServerName()))
-                    .andWith(RepeatConfig12EE8.INSTANCE.forServers(SHARED_SERVER.getServerName()));
```
[the remark this answers is not in the document]
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.config_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/CDIBuiltInConverterTest.java]: Updating the Repeat action to not be static and take a server argument

File: dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SysPropServlet.java (hunk 2 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SysPropServlet.java]: ah yes

File: dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SessionScopedFieldTestServlet.java (hunk 3 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.config_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/web/SessionScopedFieldTestServlet.java]: done

File: dev/com.ibm.ws.org.apache.cxf.3.2.jaxrs/.project [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Azquelt) [inline dev/com.ibm.ws.org.apache.cxf.3.2.jaxrs/.project]: Please remove these unrelated project files.

File: dev/com.ibm.ws.org.apache.cxf.jaxrs/.project [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Azquelt) [inline dev/com.ibm.ws.org.apache.cxf.jaxrs/.project]: Remove this file

Comment (tevans78) [reply dev/com.ibm.ws.org.apache.cxf.jaxrs/.project]: ugh stupid eclipse re-creating deleted files!

**Comment by tevans78** [comment]
#build
